src/util.rs

```diff
@@ -70,6 +93,20 @@ impl Wait {
         }
     }
 
+    pub fn run_until<F>(&self, predicate: F)
+    where
+        F: FnMut() -> bool,
+    {
+        let mut predicate = predicate;
+
+        loop {
+            if predicate() {
```

Comment:
without using the timeout this could hang forever

Reply:
i wanted to make this method for specific rare use cases such as on a slow internet the dom can render very late. but your right